src/blob/index/header.rs

```diff
@@ -0,0 +1,67 @@
+use super::prelude::*;
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub(crate) struct IndexHeader {
+    pub records_count: usize,
+    // contains serialized size of record headers, which allows to calculate offset in
+    // case of `OnDisk` state of indices
+    pub record_header_size: usize,
+    pub filter_buf_size: usize,
+    pub hash: Vec<u8>,
+    pub(crate) version: u64,
```

Comment:
@pyakushin Looks like version as u64 and written as u8 is overkill here ;) Should be it optimized now?

Reply:
I agree, we can use `u8` for both fields. 1 bit for `written` and 7 bits for version. I think we have enough versions reserved, 2^7 = 128 versions

Reply:
Changed it, need your review!